### PR TITLE
Add meta_data field to peer

### PR DESCRIPTION
**Pull request by SWvheerden**
## Description

This PR adds in meta_data field to the peer that can be used by upstream clients. 

## Motivation and Context

I used a Hashmap<u8,Vec<u8>> to ensure future proofing of the field. This will also allow more than one struct of information to be serialised into the peer. By using a serialised piece of info the peer does not need to be aware of the application layer etc.

## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [x] I'm merging against the `development` branch.
* [x] I ran `cargo-fmt --all` before pushing.
* [x] I ran `cargo test` successfully before submitting my PR.
* [ ] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

**Review by sdbondi — APPROVED** [review]
Looks good!

File: comms/src/peer_manager/migrations/v3.rs [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+use log::*;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use tari_crypto::tari_utilities::hex::serialize_to_hex;
+use tari_storage::{
+    lmdb_store::{LMDBDatabase, LMDBError},
+    IterationResult,
+};
+
+const LOG_TARGET: &str = "comms::peer_manager::migrations::v2";
```
Comment (delta1) [inline comms/src/peer_manager/migrations/v3.rs:48]: I know this is merged now, but just wanted to check if this should be `::v3` instead?

Comment (SWvheerden) [reply comms/src/peer_manager/migrations/v3.rs:48]: Yes This should be, thanks for pointing out. Fixing in my current code

Comment (delta1) [reply comms/src/peer_manager/migrations/v3.rs:48]: Nice one 👍

Comment (SWvheerden) [reply comms/src/peer_manager/migrations/v3.rs:48]: #2318

**Pull request merged**